WebAuthn: Fix credential parameters conversion to WebAuthn4j object model when used with Spring Session

ltanguy opened the pull request:
Replace comparison based on the != operator by an .equals() comparison based on the text value. The 2 objects have the same value in most setups, but not when used in conjunction with Spring Session for example, as the object in session is built for each request.

Fixes gh-17164

rwinch reviewed (changes requested):
Thanks for the pull request! Can you please:

- add a unit test for this change?
- rebase this off the oldest version of the code that contains this bug so that it is fixed for older versions? We will merge the bug forward

ltanguy commented:
Thanks for the feedback.
I rebased the branch on the oldest commit and added a unit test.
I also added a commit that makes all classes that are used by PublicKeyCredentialCreationOptions serializable, in order to be fully compatible with Spring Session that uses the Java object serializer by default.

re1709 commented:
Any update on this please? Been holding off implementing passkeys as would require the same changes made here.

ltanguy commented:
@rwinch Is there something else expected on this pull request ?
